### ISSUE TITLE
Poor metrics on a new training run using same dataset as an older run using same parameters

### ISSUE THREAD

**Issue by billdraper** [ultralytics/hub#1247]
### Search before asking

- [x] I have searched the HUB [issues](https://github.com/ultralytics/hub/issues) and [discussions](https://github.com/ultralytics/hub/discussions) and found no similar questions.

### Question

I trained a dataset today and the results were unexpectedly very poor. As an experiment to see if there is an issue with my dataset I re-trained an older dataset using the same parameters as I used when I trained from that same dataset in November, 2025 and receive dramatically different results. So, same dataset, same parameters, producing dramatically different model metrics from November 2025 and January 2026. I've attached screen snapshots of the 2 training session metrics and it includes the parameters used.

Has something significant changed with the training tools on UL Hub recently? Or am I missing something obvious? Thank you for your help.

<img width="3045" height="1653" alt="Image" src="https://github.com/user-attachments/assets/a9f4d354-a6eb-4776-8c99-b44a8f554829" />

<img width="3015" height="1638" alt="Image" src="https://github.com/user-attachments/assets/907375d3-01eb-4805-9d1b-48e310de4f0e" />

**Comment by pderrenger** [comment]
Thanks for the detailed screenshots @billdraper — with the *same dataset + “same parameters”*, you generally shouldn’t see a **dramatic** metric swing between a November 2025 run and your Jan 16, 2026 run unless something material changed in one of these buckets:

## 1) Training environment/version changed (most common across months)
Ultralytics HUB Cloud Training is backed by the Ultralytics YOLO training stack, which is updated periodically. Even if your UI parameters look identical, underlying defaults/behavior can shift across releases (optimizer defaults, augmentation internals, determinism defaults, etc.).

**Action:** please paste the first ~20 lines of the training logs from **both** runs (they usually include the `ultralytics` version, torch version, device info, etc.). That will immediately tell us if the training engine changed.

## 2) Seed / determinism / split differences
If the run is not pinned to a fixed seed (or if determinism differs), you can absolutely get noticeably different results, especially on smaller datasets.

Ultralytics YOLO exposes both `seed` and `deterministic` for reproducibility — see the arguments table in the [Train mode docs](https://docs.ultralytics.com/modes/train/) (look for `seed` and `deterministic`). If one run used different values (or defaults changed between versions), that alone can explain a big swing.

**Action:** confirm whether your HUB training “Advanced” settings included a fixed seed, and whether both runs used the same train/val split.

## 3) “Same dataset” but not the same *dataset version*
A very common gotcha is that the dataset name is the same, but the underlying artifact changed (re-upload, regenerated labels, different split, different preprocessing, etc.). Even a small label/export difference can crater mAP.

**Action:** in Ultralytics HUB, verify:
- train/val image counts match exactly
- class list/order matches exactly
- the dataset “version”/artifact shown in the run details is identical

## 4) Pretrained vs from-scratch (or different starting weights)
If one run started from pretrained weights and the other effectively trained from scratch (or from different base weights), metrics can diverge a lot.

**Action:** confirm whether `pretrained` was enabled and what the starting weights were. (Ultralytics generally recommends pretrained for best results; see `pretrained` in the [Train mode docs](https://docs.ultralytics.com/modes/train/).)

## 5) Your *new* January dataset may simply be harder (data drift)
For the dataset you trained “today” (Jan 16, 2026) that performed poorly: if its image distribution shifted (lighting, camera, backgrounds, object scale), it can behave like classic **data drift** relative to what your previous setup generalized to. See the [Data Drift glossary](https://www.ultralytics.com/glossary/data-drift) for a concrete CV-oriented explanation and mitigation ideas (collect representative data + retrain).

---

## What we need from you to pinpoint this quickly
1) Links to the two HUB training runs (if they’re private, paste the **log header** from both runs).  
2) Dataset ID/version + train/val counts for both runs.  
3) Model used (e.g., Ultralytics YOLO26n vs another variant) and whether it was **pretrained**.

Once you share those, we can tell you whether this is (a) expected reproducibility variance, (b) a dataset/split/version mismatch, or (c) something we need to fix on the HUB side.
